azdev/operations/code_gen.py

```diff
@@ -300,3 +303,64 @@ def _create_package(prefix, repo_path, is_ext, name='test', display_name=None, d
         result = pip_cmd('install -e {}'.format(new_package_path), "Installing `{}{}`...".format(prefix, name))
         if result.error:
             raise result.error  # pylint: disable=raising-bad-type
+
+
+def _get_swagger_readme_file_path(ext_name, swagger_repo):
+    from urllib import request, error
+
+    swagger_readme_file_path = None
+    if swagger_repo == const.GITHUB_SWAGGER_REPO_URL or \
+            (swagger_repo.startswith('https://') and swagger_repo.endswith(const.SWAGGER_REPO_NAME)):
+        swagger_readme_file_path = '{}/blob/master/specification/{}/resource-manager'.format(
+            swagger_repo, ext_name)
+        # validate URL
+        try:
+            request.urlopen(swagger_readme_file_path)
+        except error.HTTPError as ex:
+            raise CLIError('HTTPError: {}\nThe URL {} does not exist.'.format(ex.code, swagger_readme_file_path))
+    else:
+        swagger_readme_file_path = os.path.join(swagger_repo, 'specification', ext_name, 'resource-manager')
+        if not os.path.isdir(swagger_readme_file_path):
+            raise CLIError("The path {} does not exist.".format(swagger_readme_file_path))
+    return swagger_readme_file_path
+
+
+def _generate_extension(ext_name, repo_path, swagger_readme_file_path, use):
+    heading('Start generating extension {}.'.format(ext_name))
+    # check if npm is installed
+    try:
+        subprocess.run('npm --version', shell=True, check=True, stdout=subprocess.DEVNULL)
+    except subprocess.CalledProcessError as ex:
+        raise CLIError('{}\nPlease install npm.'.format(ex))
+    # check if autorest is installed
+    try:
+        subprocess.run(
+            'npm list -g autorest', shell=True, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError:
+        display('Installing autorest.\n')
+        try:
+            if const.IS_WINDOWS:
+                subprocess.run('npm install -g autorest', shell=True, check=True)
+            else:
+                display('May need to provide your password.\n')
+                subprocess.run('sudo npm install -g autorest', shell=True, check=True)
+        except subprocess.CalledProcessError as ex:
+            raise CLIError("Failed to install autorest.\n{}".format(ex))
+    # update autorest core
+    subprocess.check_output('autorest --latest', shell=True)
+    if not use:
```

Comment:
does use mean a specific autorest version here? If so can you make the variable name like "use_version"?

Reply:
Yes. Here is using the same cmd parameter name "use" as in autorest, do you think the parameter name should change as well?

Reply:
Oh, ya that's not a good description in autorest either, should be informative. Like "use" begs the question "use what exactly".

Reply:
The value of 'use' is not a version string, it's a release download link in Github.